### ISSUE TITLE
Unable to load rulesets from NPM package

### ISSUE THREAD

**Issue by dpopp07** [stoplightio/spectral#2072]
**Describe the bug**
According to [the documentation](https://meta.stoplight.io/docs/spectral/ZG9jOjI1MTky-sharing-and-distributing-rulesets#npm), Spectral supports loading rulesets from other NPM packages. However, this appears to be broken. When defining a ruleset with "extends", Spectral seems to only look within the local filesystem.

**To Reproduce**

1. Given any OpenAPI document and the following `.spectral.yaml` file:
```yaml
extends: '@ibm-cloud/openapi-ruleset'
```
Note: I have tried this with other public Spectral rulesets on NPM, all with the same result.

2. Run this CLI command:
```
spectral lint <your-api-definition>`
```

3. See error:
```
ENOENT: no such file or directory, open '<path-to-cwd>/@ibm-cloud/openapi-ruleset'
```

**Expected behavior**
I would expect to see the behavior described in the the documentation - Spectral would find the NPM package and retrieve the ruleset from it. Perhaps I'm reading the documentation wrong but the way I read it, installing the ruleset NPM package locally should not be required. That should just allow me to version control it but it shouldn't be necessary. It's worth noting that if I _do_ have the ruleset installed within `node_modules/` in my current directory, Spectral will find it just fine.

This is a fairly show-stopping bug as far as the configurable and extensible nature of Spectral goes (two of the tool's best assets). I seem to remember this working correctly at some point but I can't prove that to myself. I am seeing this error as far back as v6.0.0.

**Environment (remove any that are not applicable):**
 - Library version: Spectral CLI 6.2.1
 - OS: Mac

**Comment by P0lip** [comment]
Heya!

There's a gap in the documentation, but it's indeed not supposed to work for anything other than the local filesystem.
If you need to load it externally, you can use services like [unpkg.com](https://unpkg.com/).

I'll update the documentation shortly.

**Comment by dpopp07** [comment]
@P0lip Ah, okay. Thanks for the info and the prompt documentation update. It's not a requirement to load externally but it would be really nice.

In that case, I have a follow up question - is it possible to load an external, [JS formatted ruleset](https://meta.stoplight.io/docs/spectral/ZG9jOjI1MTg5-custom-rulesets#alternative-js-ruleset-format) through `unpkg`? Based on some quick tests, this isn't supported. I suppose this makes sense, since only a raw JS file is being given to Spectral and not the whole package but if I'm missing something and there's a way to do this, let me know. It would be great to be able to load a JS packaged ruleset externally.

**Comment by P0lip** [comment]
> I suppose this makes sense, since only a raw JS file is being given to Spectral and not the whole package but if I'm missing something and there's a way to do this, let me know. It would be great to be able to load a JS packaged ruleset externally.

What's the error you're getting? I feel like it might be something related to Ajv which is used by the AsyncAPI ruleset (which unfortunately is loaded even if you only use the OAS ruleset).
IIRC Skypack couldn't bundle Ajv, and unpkg might be no different here.

**Comment by dpopp07** [comment]
> What's the error you're getting?

Been playing around with this and getting different errors depending on how I package it. That said, I'm not convinced the problem isn't on my end. I'm not going to spend much more time on it and you answered my primary question so I'll close out this issue. If I ever come back to it and identify the issue is with Spectral, I'll open a new issue. Thanks again!
